beater/handlers.go

```diff
@@ -52,17 +52,63 @@ type routeMapping struct {
 	ProcessorFactory
 }
 
+type serverResponse struct {
+	err     error
+	code    int
+	counter *monitoring.Int
+}
+
 var (
-	serverMetrics  = monitoring.Default.NewRegistry("apm-server.server")
-	requestCounter = monitoring.NewInt(serverMetrics, "requests.counter")
-	responseValid  = monitoring.NewInt(serverMetrics, "response.valid")
-	responseErrors = monitoring.NewInt(serverMetrics, "response.errors")
+	serverMetrics = monitoring.Default.NewRegistry("apm-server.server")
+	counter       = func(s string) *monitoring.Int {
+		return monitoring.NewInt(serverMetrics, s)
+	}
```

Comment:
I'd prefer adding a `newServerResponse(err error, code int, name string)` method that abstracts how the counter gets created.

Reply:
with a `new` function how do you do the ones that take an error from the processor (eg cannotDecode), you cant create a counter in  `newServerResponse` because there can only be one monitor instance with the given name, so you have to pass it from the outside. you know what i mean?

Reply:
that's true, would be more difficult